RFC: nixos/home-assistant: NixOS is an unsupported installation method

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Trying to steer NixOS users away from reporting bugs to the upstream,
when they don't have the capacity to support bugs that could be the
result of our downstreaming setup.

Probably not the perfect place to put this, but I couldn't think of a
better one yet.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt committed:
nixos/home-assistant: NixOS is an unsupported installation method

Trying to steer NixOS users away from reporting bugs to the upstream,
when they don't have the capacity to support bugs that could be the
result of our downstreaming setup.

fabaff commented:
In the meantime it's also mentioned in the wiki: https://nixos.wiki/wiki/Home_Assistant

mweinelt committed:
home-assistant: evaluate outdated dependencies based on our package set

Home Assistant overwrites a bunch of dependencies resulting in an
overridden python package set. Expose that via passthru as
`home-assistant.python.pkgs` and use that in parse-requirements.py,
so we take overridden versions into consideration.

mweinelt commented:
Also added a section concerning how to run the (downstream-supported) Home Assistant version from NixOS unstable. https://nixos.wiki/wiki/Home_Assistant#Running_a_recent_version_using_an_overlay

And squashed a bug in our dependency evaluation that would make it not consider overwritten package versions.

dotlambda commented on pkgs/servers/home-assistant/default.nix:
```diff
@@ -799,6 +799,7 @@ in with py.pkgs; buildPythonApplication rec {
     tests = {
       inherit (nixosTests) home-assistant;
     };
+    pythonPackages = py.pkgs;
```
```suggestion
    python = py;
```

mweinelt replied:
What would be the benefit of exposing all of it?

dotlambda replied:
I guess it's personal preference but we override `python`, not `pythonPackages`, so I think that should be the thing in `passthru`.

mweinelt replied:
WFM.

dotlambda commented on pkgs/servers/home-assistant/parse-requirements.py:
```diff
@@ -32,7 +32,7 @@
 
 COMPONENT_PREFIX = "homeassistant.components"
-PKG_SET = "python3Packages"
+PKG_SET = "home-assistant.pythonPackages"
```
```suggestion
PKG_SET = "home-assistant.python.pkgs"
```

dotlambda commented:
Now the commit message needs to be corrected though.

mweinelt commented:
> Now the commit message needs to be corrected though.

Updated, thanks.

dotlambda commented:
Some logic in `parse-requirements.py` must be wrong if this actually changes the number of supported components.
